# Newton's Rings

In [1]:
import numpy as np
import pandas as pd

In [12]:
wavelength = 589.3 * 10**(-9)
def Radius(r2,r1):
    return (r2**2 - r1**2)/(4*wavelength)

In [21]:
reading = pd.read_csv("Newton's rings - Sheet1.csv",usecols= [3,8],header = 0).dropna()
k = reading.keys()
read1 = np.array(reading[k[0]])
read2 = np.array(reading[k[1]])

diameter = abs(read2-read1)

R_c = [Radius(diameter[i],diameter[i+1]) for i in range(0,20,2)]
R_avg = np.average(R_c)
R_avg

1.1350330901069063

In [23]:
del_D = (0.5/50)*10**(-3)
l = np.array([del_D * (diameter[i] - diameter[i+1]) / (2 * wavelength) for i in range(0,20,2)])
maxerr = max(l)
maxerr/R_avg

0.005456923939450595

 The Radius of Curvature of the lens is $R~=~~1.1350330901069063$
 
 The maximum permissible error is  $\delta R ~=~~ 0.005456923939450595$